### PR TITLE
Ethernet over USB docs and helper scripts

### PR DESCRIPTION
**Pull request by adamfeuer**
### Summary

* Adds defconfig for Ethernet over USB 2.0 High Speed that works reliably
* Removes other redundant Ethernet over USB defconfigs
* Improves Ethernet over USB helper script and moves it to tools
* Updates SAMA5D3-Xplained README.txt to cover setup and usage

### Impact

* None if you don't use the CDC ECM driver
* CDC ECM driver can now do 4.4 MBytes/s (34 Mbits/s) on the SAMA5D3-Xplained

### Limitations / TODO

* Reliability seems to depend on correct IOB configuration. This one was arrived at via trial and error.

### Testing

* Via an improved tcpblaster, this will be submitted in another PR

### How To Verify 

* Set up via the README.txt instructions and use tcpblaster to verify speed and ability to run for long times.

### CODE REVIEW

File: boards/arm/sama5/sama5d3-xplained/README.txt [hunk 1]
```diff
@@ -834,6 +836,21 @@ Networking
...
+
+    CONFIG_USBDEV=y
+    CONFIG_USBDEV_DMA=y
+    CONFIG_USBDEV_DUALSPEED=y
+    CONFIG_NET_CDCECM=y
+    CONFIG_NET_ETH_PKTSIZE=1514
+
+  You can also use the defconfig file in `boards/arm/sama5/sama5d3-xplained/configs/ethernet-over-usb-2-high-speed`.
+
+
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d3-xplained/README.txt]: Remove the redundant blank line.

File: boards/arm/sama5/sama5d3-xplained/README.txt [hunk 2]
```diff
@@ -1019,6 +1036,74 @@ Networking
...
+  $ telnet 10.0.0.2
+
+  The helper script also sets up Network Address Translation (NAT) so the NuttX system
+  can access the Internet. If that is not what you want, you can remove the iptables
+  NAT commands from the script.
+
+
+
+
+
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d3-xplained/README.txt]: Remove the redundant blank line too.

File: tools/netusb.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    ip route show
+    ping -c 1 $IP_NUTTX
+else
+    net_off
+fi
+
+
+
+
+
```
Comment (xiaoxiang781216) [inline tools/netusb.sh]: Remove the redundant blank line.

File: tools/netusb.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  echo "Usage: $0 <main-interface> <usb-net-interface> <on|off>"
+  exit 1
+fi
+
+IF_HOST=$1
+IF_USB=$2
+STATUS=$3
+
+net_off() {
+    ip route delete $IP_NET
```
Comment (xiaoxiang781216) [inline tools/netusb.sh]: use two spaces indent

File: tools/netusb.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ifconfig $IF_USB down
+}
+
+if [ "$STATUS" == "on" ]; then
+    net_off
+    ifconfig $IF_USB up
+    ifconfig -a
+    ifconfig $IF_USB add $IP_HOST
+    ifconfig $IF_USB:0 broadcast $IP_BROADCAST netmask $IP_NETMASK
+    ip route delete $IP_NET
```
Comment (xiaoxiang781216) [inline tools/netusb.sh]: Remove, done in net_off

File: tools/netusb.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -x
+
+# USB Ethernet Gadget interface
```
Comment (xiaoxiang781216) [inline tools/netusb.sh]: Add some description in tools/README.txt

File: tools/netusb.sh [hunk 7]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
```
Comment (xiaoxiang781216) [inline tools/netusb.sh]: Add apache copyright, like tools/zds/zdsar.c

File: boards/arm/sama5/sama5d3-xplained/README.txt [hunk 8]
```diff
@@ -1019,6 +1037,75 @@ Networking
       CONFIG_NSH_NETINIT_RETRYMSEC=2000     : Configure the network monitor as you like
       CONFIG_NSH_NETINIT_SIGNO=18
 
+
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d3-xplained/README.txt]: Remove the redundant blank line.

File: boards/arm/sama5/sama5d3-xplained/README.txt (hunk 1 above)
Comment (adamfeuer) [reply boards/arm/sama5/sama5d3-xplained/README.txt]: @xiaoxiang781216 will fix

File: tools/netusb.sh (hunk 3 above)
Comment (adamfeuer) [reply tools/netusb.sh]: @xiaoxiang781216 will fix

File: tools/netusb.sh (hunk 4 above)
Comment (adamfeuer) [reply tools/netusb.sh]: @xiaoxiang781216 will fix

File: tools/netusb.sh (hunk 6 above)
Comment (adamfeuer) [reply tools/netusb.sh]: @xiaoxiang781216 will fix

File: tools/netusb.sh (hunk 7 above)
Comment (adamfeuer) [reply tools/netusb.sh]: @xiaoxiang781216 will fix

File: boards/arm/sama5/sama5d3-xplained/README.txt (hunk 8 above)
Comment (adamfeuer) [reply boards/arm/sama5/sama5d3-xplained/README.txt]: @xiaoxiang781216 will fix

File: tools/netusb.sh (hunk 5 above)
Comment (adamfeuer) [reply tools/netusb.sh]: @xiaoxiang781216 Got it, will fix.

**Comment by adamfeuer** [comment]
@xiaoxiang781216 Thanks for the improvements. It's night here in Seattle, I'll make the changes you suggested tomorrow. 🙂

File: boards/arm/sama5/sama5d3-xplained/README.txt (hunk 2 above)
Comment (adamfeuer) [reply boards/arm/sama5/sama5d3-xplained/README.txt]: @xiaoxiang781216 will fix

**Comment by adamfeuer** [comment]
@xiaoxiang781216 I made the requested improvements, rebased, and force-pushed the branch. Will you check it out to see if there are more changes you'd like to see?

**Pull request merged**